drivers/soundwire/stream.c

```diff
@@ -1752,17 +1789,31 @@ int sdw_deprepare_stream(struct sdw_stream_runtime *stream)
 {
 	int ret;
 
+	pr_err("%s: %s: start\n", __func__, stream->name);
+
 	if (!stream) {
 		pr_err("SoundWire: Handle not found for stream\n");
 		return -EINVAL;
 	}
 
 	sdw_acquire_bus_lock(stream);
+
+	if (stream->state != SDW_STREAM_DISABLED) {
```

Comment:
SDW_STREAM_PREPARED is also valid here, right? We may do prepare -> deprepare without enabling/disabling.